### ISSUE TITLE
Difficult to understand the error message when the extension is missing

### ISSUE THREAD

**Issue by taiju** [denoland/deno#8024]
I found it difficult to understand the error message when the script file was created without the extension.

## To Reproduce

```sh
deno --version
deno 1.4.6
v8 8.7.220.3
typescript 4.0.3
```

```sh
rm -rf ~/.cache/deno/*
```
```sh
cat <<EOT > mod
import * as log from "https://deno.land/std/log/mod.ts";

log.info("hello");
EOT
```
```sh
deno run mod
```

### Result

```
thread 'main' panicked at 'Cached source file doesn't exist', cli/global_state.rs:222:8
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

If there is no import statement, it will not be reproduced. (So it was even more confusing.)

It would be nice if we could read the error messages to see that the files specified in the deno run must have an extension.

**Comment by grosto** [comment]
I think it makes sense to throw if some module has no/unsupported media type in the dependency graph.

I can work on this if maintainers agree with this behaviour.

**Comment by kitsonk** [comment]
@grosto this area of code is going under a major refactor (see: #7225 and #7996).  I plan to try to address this already as part of #7996.

Local files without extensions should actually work and be treated as JavaScript, so I am not totally sure why this is currently failing.

**Comment by grosto** [comment]
> Local files without extensions should actually work and be treated as JavaScript, so I am not totally sure why this is currently failing.

I think it's failing because ModuleGraph does not extract and fetch dependencies for files other than js, jsx, ts, tsx.

**Comment by kitsonk** [comment]
This is partially fixed in #7996 but I will need to revisit it to make sure we can support it fully.  There are challenges with `tsc` with extension-less root modules that need to be addressed.
